ui/src/organizations/components/VariableList.tsx

```diff
@@ -35,13 +53,45 @@ class VariableList extends PureComponent<Props> {
                 key={`variable-${variable.id}`}
                 variable={variable}
                 onDeleteVariable={onDeleteVariable}
+                onUpdateVariable={onUpdateVariable}
+                onEditVariable={this.handleStartEdit}
               />
             ))}
           </IndexList.Body>
         </IndexList>
+        <OverlayTechnology visible={this.isVariableOverlayVisible}>
+          <UpdateVariableOverlay
+            variable={this.variable}
+            onCloseOverlay={this.handleCloseOverlay}
+            onUpdateVariable={this.handleUpdateVariable}
+          />
+        </OverlayTechnology>
       </>
     )
   }
+
+  private get variable(): Variable {
+    return this.props.variables.find(v => v.id === this.state.variableID)
+  }
+
+  private get isVariableOverlayVisible(): boolean {
+    return this.state.variableOverlayState === OverlayState.Open
+  }
+
+  private handleCloseOverlay = () => {
+    this.setState({variableOverlayState: OverlayState.Closed})
```

Comment:
should this also reset variableID to `null`?

Reply:
good catch!